### PR TITLE
Attach file IDs to `Span`s

### PR DESCRIPTION
**Pull request by g-r-a-n-t**
### What was wrong?

https://github.com/ethereum/fe/blob/bfeffe82b4f8623330c1ebb1950ca981449c9e86/crates/common/src/diagnostics.rs#L8-L11

We also can't get a list of diagnostics with labels pointing to different source files.

### How was it fixed?

Refactored things.

### To-Do

[//]: # (Stay ahead of things, add list items here!)
- [ ] OPTIONAL: Update [Spec](https://github.com/ethereum/fe/blob/master/docs/src/spec/index.md) if applicable
- [x] Add entry to the [release notes](https://github.com/ethereum/fe/blob/master/newsfragments/README.md) (may forgo for trivial changes)
- [x] Clean up commit history

### CODE REVIEW

File: crates/parser/src/ast.rs [hunk 1]
```diff
@@ -18,7 +18,7 @@ pub enum ModuleStmt {
     Use(Node<Use>),
     TypeAlias(Node<TypeAlias>),
     Contract(Node<Contract>),
-    Constant(Node<ConstantDecl>),
+    Constant(Box<Node<ConstantDecl>>),
```
Comment (g-r-a-n-t) [inline crates/parser/src/ast.rs:21]: clippy was complaining about this variant being ~2x the size of the next largest

File: crates/analyzer/tests/snapshots/analysis__call_statement_with_args.snap [hunk 2]
```diff
@@ -115,14 +115,11 @@ note:
...
-  │         ^^^^^^^^^^^ attributes hash: 8420354620523897173
+  │         ^^^^^^^^^^^ attributes hash: 8522453005002335674
   │
   = SelfAttribute {
         func_name: "assign",
-        self_span: Span {
-            start: 137,
-            end: 141,
-        },
+        self_span: 137..141,
```
Comment (sbillig) [inline crates/analyzer/tests/snapshots/analysis__call_statement_with_args.snap:122]: Yes! Why didn't we think of this earlier?

**Review by sbillig — APPROVED** [review]
Looks good to me.

File: crates/analyzer/tests/snapshots/analysis__call_statement_with_args.snap (hunk 2 above)
Comment (g-r-a-n-t) [reply crates/analyzer/tests/snapshots/analysis__call_statement_with_args.snap:122]: :man_shrugging:

**Pull request merged**